sdks/python/apache_beam/transforms/combiners.py

```diff
@@ -243,6 +260,92 @@ def SmallestPerKey(pcoll, n, reverse=True):
     return pcoll | Top.PerKey(n, reverse=True)
 
 
+class _ComparableValue(object):
+
+  __slots__ = ('value', 'less_than')
+
+  def __init__(self, value, less_than):
+    self.value = value
+    self.less_than = less_than
+
+  def __lt__(self, other):
+    return self.less_than(self.value, other.value)
+
+  def __repr__(self):
+    return "_ComparableValue[%s]" % self.value
+
+
+@with_input_types(T)
+@with_output_types(KV[None, List[T]])
+class _TopPerShard(core.DoFn):
```

Comment:
Perhaps _TopPerBundle (and MergeTopPerBundle respectively)?

Reply:
Good call.